services/provider/server/server.go

```diff
@@ -5,43 +5,202 @@ import (
 	"fmt"
 	"net"
 
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
 )
 
-type ocsProviderServer struct {
+type OCSProviderServer struct {
 	pb.UnimplementedOCSProviderServer
+	consumerAPI APIUpdater
+	authManager *AuthManager
+	namespace   string
 }
 
-// OnBoardConsumer RPC call to onboard a new OCS consumer cluster.
-func (c *ocsProviderServer) OnBoardConsumer(ctx context.Context, req *pb.OnBoardConsumerRequest) (*pb.OnBoardConsumerResponse, error) {
-	return &pb.OnBoardConsumerResponse{}, nil
+func NewOCSProviderServer(cClient APIUpdater, auth *AuthManager, namespace string) *OCSProviderServer {
+	return &OCSProviderServer{
+		consumerAPI: cClient,
+		authManager: auth,
+		namespace:   namespace,
+	}
+}
+
+// GenerateToken RPC call to generate a new jwt token for the consumer cluster
+func (s *OCSProviderServer) GenerateToken(ctx context.Context, req *pb.GenerateTokenRequest) (*pb.GenerateTokenResponse, error) {
+	token, err := s.authManager.GetAccessToken()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to generate access token: %v", err)
+	}
+	return &pb.GenerateTokenResponse{AccessToken: token}, nil
+}
+
+// OnboardConsumer RPC call to onboard a new OCS consumer cluster.
+func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.OnboardConsumerRequest) (*pb.OnboardConsumerResponse, error) {
+	// validate capacity
+	capacity, err := resource.ParseQuantity(req.Capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "%q is not a valid resource capacity: %v", req.Capacity, err)
```

Comment:
I think this should not be an codes.Internal error code

Reply:
Right.  Passing `InvalidArgument` error code.